client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -573,6 +597,21 @@ export default {
         this.toastErrorHandler(this.$t('notification:page.duplicateFailed'))
       }
     },
+
+    checkBlocksForUnsavedChanges (next) {
+      if (this.hasUpdatedBlocks) {
+        const okMessage = window.confirm(this.$t('build.unsavedChanges'))
```

Comment:
next(window.confirm(this.$t('build.unsavedChanges'))) simplifies all of this, please use it